File: repl/adrianus/python3/vm/instructions.py
class Instructions:
    size = 0

    def __init__(self):
        # a list of bytes
        # List<Byte>
        self.bytes = []

    def get(self, i):
        return self.bytes[i]

    def add(self, ins):
        self.size += 1
        self.bytes.append(ins)

    def __str__(self):
        out = ""
        for instruction in self.bytes:
            out += str(instruction) + "\n"
        return out

File: repl/adrianus/cpp-v2/vm/instructions.cpp
#include "instructions.h"

Instructions::Instructions() {
    size = 0;
}

unsigned char Instructions::get(int i) {
    return bytes.at(i);
}

void Instructions::add(unsigned char val) {
    bytes.push_back(val);
    size++;
}

void Instructions::removeLast() {
    bytes.pop_back();
    size--;
}